packages/cli/main.js

```diff
@@ -1,27 +1,49 @@
-/* eslint-disable no-console */
-
 import getopts from "getopts";
 import hrp from "http-request-plus";
 import { format, parse } from "json-rpc-protocol";
 import { mapValues, omit } from "lodash";
+import WebSocket from "ws";
 
 async function main() {
-  const opts = getopts(process.argv.slice(2));
-  const method = opts._[0];
-  const params = mapValues(omit(opts, "_"), String);
+  const args = process.argv.slice(2);
+  const { watch } = getopts(args, {
+    stopEarly: true,
+    boolean: ["watch"],
+  });
 
-  const response = parse(
-    await hrp
-      .post("http://localhost:4000/api/", {
-        body: format.request(0, method, params),
-      })
-      .readAll("utf-8")
-  );
+  if (watch) {
+    const websocket = new WebSocket("ws://localhost:4000");
 
-  if (response.type === "response") {
-    console.log(response.result);
-  } else if (response.type === "error") {
-    console.error(response.error);
+    websocket.onerror = event => {
+      console.error(event);
+    };
+    websocket.onmessage = event => {
+      const message = parse(event.data);
+      if (message.type !== "notification") {
+        return;
+      }
+      console.log(
+        "method: %s\n %s\n",
+        message.method,
+        JSON.stringify(message.params)
+      );
+    };
+  } else {
+    const opts = getopts(args);
```

Comment:
Here you should parse only the options not taken care of by the previous `getopts`:

```js
const { watch, _: optionsLeft } = getopts(/* ... */)
const opts = getopts(/* ... */)
```